internal/pkg/service/buffer/worker/distribution/listener.go

```diff
@@ -65,8 +65,8 @@ func newListeners(proc *servicectx.Process, clock clock.Clock, logger log.Logger
 				v.trigger()
 				// Stop all listeners
 				for _, l := range v.listeners {
-					l.wg.Wait()
 					l.cancel()
+					l.wg.Wait()
```

Comment:
Typo, prejavilo sa v novych testoch .... resp. ked som v kode zabudol `defer listener.Stop()`, tak sa mali pri shutdown aj tak killnut vsetky listeners, ale namiesto toho sa to seklo.